esi/src/expression.rs

```diff
@@ -128,52 +149,74 @@ impl EvalContext {
         self.request = request;
     }
 }
+
 impl<const N: usize> From<[(String, Value); N]> for EvalContext {
     fn from(data: [(String, Value); N]) -> Self {
         Self::new_with_vars(HashMap::from(data))
     }
 }
 
-fn format_key(key: &str, subkey: Option<String>) -> String {
+fn format_key(key: &str, subkey: Option<&str>) -> String {
     subkey.map_or_else(|| key.to_string(), |subkey| format!("{key}[{subkey}]"))
 }
-
-#[derive(Debug, Clone, PartialEq)]
+/// Represents a value in an ESI expression.
+///
+/// Values can be of different types:
+/// - `Integer`: A 32-bit signed integer
+/// - `String`: A UTF-8 string
+/// - `Boolean`: A boolean value (true/false)
+/// - `Null`: Represents an absence of value
+#[derive(Debug, Clone, PartialEq, Eq)]
 pub enum Value {
     Integer(i32),
-    String(String),
-    Boolean(BoolValue),
+    Text(Cow<'static, str>),
```

Comment:
Why the rename of String to Text?

Reply:
The `String` is  renamed to `Text` because it's `(Cow<'static, str>)` and not `String`... I think it's just an arbitrary name. Naming is hard :)